### ISSUE TITLE
[iOS 10] code push not updating

### ISSUE THREAD

**Issue by rogerkerse** [Microsoft/react-native-code-push#521]
Hey,

I integrated code push to my app. It worked nicely with iOS 9 and swift older version. Now I updated my project to Swift 3 and I am running it on iOS 10 and once i resume my app, I get no code push updates nor will it log anything out like it is checking for updates.

**Comment by silhouettes** [comment]
We will have a look at this within the next two days!

**Comment by rogerkerse** [comment]
Figured out the problem. I have set it to 
```
const codePushOptions = {
	checkFrequency: codePush.CheckFrequency.ON_APP_RESUME,
	installMode: codePush.InstallMode.ON_NEXT_RESUME
};
```

Immediate app resumes do not trigger codepush for some reason. Once the app has been in the background for more than few minutes, then it triggers.

**Comment by silhouettes** [comment]
Thank you, this extra information is very valuable. When you say 'trigger codepush', are you talking about the checking for the update, or the installing of it once the update has already been downloaded?

**Comment by rogerkerse** [comment]
I am talking about checking for the update.

**Comment by 15071304899** [comment]
hi~  I also meet this problem .   my project alert  an  updateDialog .   But nothing to happen after click 'continue'   ....

**Comment by silhouettes** [comment]
@rogerkerse, @15071304899: I am investigating this now, but just to make things a little easier for me, what versions of react-native and react-native-code-push are you using?

**Comment by silhouettes** [comment]
@rogerkerse, @15071304899 - just following up, can you let me know the versions of `react-native` and `react-native-code-push` you are using?

**Comment by silhouettes** [comment]
I set up an app with:

- react-native-code-push 1.14.6-beta
- react-native 0.33.0
- iOS 10.0 (simulator)
- Both immediate and resume-based checks/installs

And I wasn't able to reproduce any problems. With the resume-based, it did the check/download on the first resume, and then loaded the updated app on the second resume. With the update dialog-based install, it immediately applied the update.

@rogerkerse, @15071304899 - if you're still running into this problem, can you please provide more detail? Perhaps you could try adding more logging - a good example of how to do this is here: https://github.com/Microsoft/react-native-code-push/blob/master/Examples/CodePushDemoApp/demo.js

**Comment by silhouettes** [comment]
@15071304899: are you loading from the packager by any chance? Do you have this line in your code: `jsCodeLocation = [CodePush bundleURL];` in your AppDelegate.m, and have you tried running in release mode?

**Comment by silhouettes** [comment]
@rogerkerse, @15071304899: I am finding that the update is being downloaded, but not applied, when running against the pending changes in our master branch (our `1.14.6-beta` release is fine though). Please let me know if you saw this when running against master.

**Comment by davidsalib** [comment]
I'm having an issue where on the simulator I get a dialog to update, however on my device (in Production) I'm not getting a dialog. I'm using IMMEDIATE for updates. Is that broken on iOS 10? I tried restarting the app more than once and it doesn't detect the update.

**Comment by silhouettes** [comment]
@davidsalib, and everyone else on this thread - which versions of `react-native` and `react-native-code-push` are you running? It is extremely difficult to address your issues without this information!

**Comment by silhouettes** [comment]
@rogerkerse, @15071304899: If you were running the version of `react-native-code-push` that was on master (as opposed to a released npm version), then I am almost certain that it is fixed with #543. If this is true, and you'd like to test it, you'll need to clean install your npm dependencies and do a clean build. Please let me know if that fixed it for you!

**Comment by silhouettes** [comment]
This fix is now available in `1.15.0-beta`. Release notes can be found at: https://github.com/Microsoft/react-native-code-push/releases/tag/v1.15.0-beta. Thanks so much for the patience, and we look forward to your feedback! Please let us know if you still run into any issues.
